contracts/extensions/BasicIssuanceExtension.sol

```diff
@@ -0,0 +1,287 @@
+     * @param _maxManagerFee                Maximum fee that can be charged on issue and redeem
+     * @param _managerIssueFee              Fee to charge on issuance
+     * @param _managerRedeemFee             Fee to charge on redemption
+     * @param _feeRecipient                 Address to send fees to
+     * @param _managerIssuanceHook          Instance of the contract with the Pre-Issuance Hook function
+     */
+    function initializeModule(
+        IDelegatedManager _delegatedManager,
+        uint256 _maxManagerFee,
+        uint256 _managerIssueFee,
+        uint256 _managerRedeemFee,
+        address _feeRecipient,
+        address _managerIssuanceHook
+    ) 
+        external 
+        onlyValidManager(_delegatedManager) 
+    {
+        require(msg.sender == _delegatedManager.owner(), "Must be owner");
+        require(_delegatedManager.isInitializedExtension(address(this)), "Extension must be initialized");
+
+        ISetToken setToken = _delegatedManager.setToken();
+
+        bytes memory callData = abi.encodeWithSignature(
+            "initialize(address,uint256,uint256,uint256,address,address)", 
+            setToken,
+            _maxManagerFee,
+            _managerIssueFee,
+            _managerRedeemFee,
+            _feeRecipient,
+            _managerIssuanceHook
+        );
+        _invokeManager(setToken, address(issuanceModule), callData);
+
+        BasicIssuanceModuleInitialized(address(setToken), address(_delegatedManager));
+    }
+
+    /**
+     * ONLY OWNER: Initializes BasicIssuanceExtension to the DelegatedManager.
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeExtension(IDelegatedManager _delegatedManager) external onlyValidManager(_delegatedManager) {
+        ISetToken setToken = _delegatedManager.setToken();
+
+        require(msg.sender == _delegatedManager.owner(), "Must be owner");
+        require(_delegatedManager.isPendingExtension(address(this)), "Extension must be pending");
+
+        setManagers[setToken] = _delegatedManager;
+
+        _delegatedManager.initializeExtension();
+
+        BasicIssuanceExtensionInitialized(address(setToken), address(_delegatedManager));
+    }
+
+    /**
+     * ONLY OWNER: Initializes BasicIssuanceExtension to the DelegatedManager and BasicIssuanceModule to the SetToken
+     *
+     * @param _delegatedManager             Instance of the DelegatedManager to initialize
+     * @param _maxManagerFee                Maximum fee that can be charged on issue and redeem
+     * @param _managerIssueFee              Fee to charge on issuance
+     * @param _managerRedeemFee             Fee to charge on redemption
+     * @param _feeRecipient                 Address to send fees to
+     * @param _managerIssuanceHook          Instance of the contract with the Pre-Issuance Hook function
+     */
+    function initializeModuleAndExtension(
```

Comment:
Can we refactor this to use the code in `initializeModule` and `initializeExtension`?

Reply:
Could probably even add the `initializeExtension` one to `BaseGlobalExtension`

Reply:
> Can we refactor this to use the code in `initializeModule` and `initializeExtension`?

Yes we would have to 
- make `initializeModule` and `initializeExtension` public
- we would hit `require(msg.sender == _delegatedManager.owner(), "Must be owner");` twice and `require(_delegatedManager.isInitializedExtension(address(this)), "Extension must be initialized");` in `initializeModule` would be redundant 



> Could probably even add the `initializeExtension` one to `BaseGlobalExtension`

Currently all `initializeExtension` are the same so this would make sense. Only thing is that we have specific event names (`TradeExtensionInitialized`). We can pass the extension address as an event parameter though

---

Also we should add a `ModuleInitialized` event to `initializeModuleAndExtension`. This would happen if we use `initializeModule` within.